CMakeLists.txt

```diff
@@ -37,7 +37,14 @@ find_package (PythonInterp REQUIRED)
 find_package (FLEX REQUIRED)
 find_package (BISON REQUIRED)
 find_package (Protobuf 3.0.0 REQUIRED)
+# The boost graph headers are optional and only required by the graphs backend
```

Comment:
It's not really worth making it optional, IMO, but since you've already done the work it's up to you.